flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/serialization/SpanningRecordSerializerTest.java

```diff
@@ -40,25 +41,25 @@
 
 	@Test
 	public void testHasSerializedData() throws IOException {
-		final int segmentSize = 16;
-
 		final SpanningRecordSerializer<SerializationTestType> serializer = new SpanningRecordSerializer<>();
 		final SerializationTestType randomIntRecord = Util.randomRecord(SerializationTestTypeFactory.INT);
 
 		Assert.assertFalse(serializer.hasSerializedData());
 
-		serializer.addRecord(randomIntRecord);
+		serializer.serializeRecord(randomIntRecord);
 		Assert.assertTrue(serializer.hasSerializedData());
 
-		serializer.continueWritingWithNextBufferBuilder(createBufferBuilder(segmentSize));
+		final BufferBuilder bufferBuilder1 = createBufferBuilder(16);
+		serializer.copyToBufferBuilder(bufferBuilder1);
 		Assert.assertFalse(serializer.hasSerializedData());
 
-		serializer.continueWritingWithNextBufferBuilder(createBufferBuilder(8));
-
-		serializer.addRecord(randomIntRecord);
+		final BufferBuilder bufferBuilder2 = createBufferBuilder(8);
+		serializer.serializeRecord(randomIntRecord);
+		serializer.copyToBufferBuilder(bufferBuilder2);
 		Assert.assertFalse(serializer.hasSerializedData());
 
-		serializer.addRecord(randomIntRecord);
+		serializer.serializeRecord(randomIntRecord);
```

Comment:
why don't you use `serializer.reset()` here? (serialize only once as in production code)